engine/lib/filestore.php

```diff
@@ -517,6 +517,11 @@ function _elgg_filestore_detect_mimetype($hook, $type, $mime_type, $params) {
 	if ($mime_type == "application/vnd.ms-office" && $info['extension'] == "ppt") {
 		$mime_type = "application/vnd.ms-powerpoint";
 	}
+	
+	// hack for browser providing better mime-type than server can detect
+	if ($mime_type == "application/octet-stream" && preg_match('~^(audio|image|video)/~', $params['default'])) {
```

Comment:
I'm not sure I feel comfortable just trusting the client to give us the right type. I was expecting we would set from extension based on a simple set of ext => mime mappings.